packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -32,27 +33,31 @@ export default class OperatorModeStateService extends Service {
 
   addItemToStack(item: StackItem, stackIndex = 0) {
     this.state.stacks[stackIndex].items.push(item);
-    this.persist();
+    this.schedulePersist();
   }
 
   removeItemFromStack(item: StackItem, stackIndex = 0) {
     let itemIndex = this.state.stacks[stackIndex].items.indexOf(item);
     this.state.stacks[stackIndex].items.splice(itemIndex);
-    this.persist();
+    this.schedulePersist();
   }
 
   replaceItemInStack(item: StackItem, newItem: StackItem, stackIndex = 0) {
     let itemIndex = this.state.stacks[stackIndex].items.indexOf(item);
     this.state.stacks[stackIndex].items.splice(itemIndex, 1, newItem);
-    this.persist();
+    this.schedulePersist();
   }
 
   clearStack(stackIndex = 0) {
     this.state.stacks[stackIndex].items.splice(0);
-    this.persist();
+    this.schedulePersist();
   }
 
-  persist() {
+  private schedulePersist() {
+    scheduleOnce('afterRender', this, this.persist);
```

Comment:
```suggestion
    // When multiple stack manipulations are bunched together in a loop, for example when closing multiple cards in a loop,
    // we get into a async race condition where the change to cardController.operatorModeState will reload the route and
    // restore the state from the query param in a way that is out of sync with the state in the service. To avoid this,
    // we do the change to the query param only after all modifications to the state have been rendered.
    scheduleOnce('afterRender', this, this.persist);
```